sos/report/plugins/microovn.py

```diff
@@ -0,0 +1,152 @@
+                        'ovn-northd.pid', 'ovn-controller.pid']:
+            self.add_copy_spec([
+                self.path_join('/var/snap/microovn/common/run/ovn', pidfile),
+            ])
+
+        ovn_controller_sock_path = self._find_sock(
+            self.ovn_sock_path, self.ovn_controller_sock_regex)
+
+        northd_sock_path = self._find_sock(self.ovn_sock_path,
+                                           self.ovn_northd_sock_regex)
+
+        # ovsdb nb/sb cluster status commands
+        cst = "cluster/status"
+        cos = "connection-status"
+        pfx = "microovn."
+        ovn_nb = "OVN_Northbound"
+        ovn_sb = "OVN_Southbound"
+        self.add_cmd_output([
+            f"{pfx}ovs-appctl -t {self.ovn_nbdb_sock_path} {cst} {ovn_nb}",
+            f"{pfx}ovs-appctl -t {self.ovn_sbdb_sock_path} {cst} {ovn_sb}",
+            f"{pfx}ovn-appctl -t {northd_sock_path} status",
+            f"{pfx}ovn-appctl -t {ovn_controller_sock_path} {cos}",
+        ], foreground=True, timeout=30
+        )
+
+        # Some user-friendly versions of DB output
+        nbctl_cmds = [
+            '{pfx}ovn-nbctl --no-leader-only show',
+            '{pfx}ovn-nbctl --no-leader-only get-ssl',
+            '{pfx}ovn-nbctl --no-leader-only get-connection',
+        ]
+
+        sbctl_cmds = [
+            '{pfx}ovn-sbctl --no-leader-only show',
+            '{pfx}ovn-sbctl --no-leader-only lflow-list',
+            '{pfx}ovn-sbctl --no-leader-only get-ssl',
+            '{pfx}ovn-sbctl --no-leader-only get-connection',
+        ]
+
+        # Database Output
+        nb_tables = self.get_tables_from_schema(self.path_join(
+            '/snap/microovn/current/share/ovn', 'ovn-nb.ovsschema'))
+        self.add_database_output(nb_tables, nbctl_cmds,
+                                 'microovn.ovn-nbctl --no-leader-only')
+        cmds = nbctl_cmds
+
+        sb_tables = self.get_tables_from_schema(self.path_join(
+            '/snap/microovn/current/share/ovn',
+            'ovn-sb.ovsschema'), ['Logical_Flow'])
+        self.add_database_output(sb_tables, sbctl_cmds,
+                                 'microovn.ovn-sbctl --no-leader-only')
+        cmds += sbctl_cmds
+
+        # If OVN is containerized, we need to run the above commands inside
+        # the container. Removing duplicates (in case there are) to avoid
+        # failing on collecting output to file on container running commands
+        cmds = list(set(cmds))
+        self.add_cmd_output(
+            cmds, foreground=True
+        )
+
+        for dbfile in ["ovnnb_db.db", "ovnsb_db.db"]:
+            path = '/var/snap/microovn/common/data/central/db'
+            dbfilepath = self.path_join(path, dbfile)
+            if os.path.exists(dbfilepath):
+                self.add_copy_spec(dbfilepath)
+                self.add_cmd_output(
+                    f"ls -lan {dbfilepath}", foreground=True)
+
+        self.add_journal(units="ovn-northd")
```

Comment:
This is collected by the `ovn_central` plugin today. Is there a microovn-specific variant for this unit? Does the `ovn_central` plugin not get enabled in this kind of environment?

Reply:
MicroOVN essentially implements ovn_central in a snap. The regular ovn_central plugin does not get triggered in these environments. I initially thought combining this with the existing ovn_central, but then decided it would be cleaner as a separate plugin similar to all other micro* plugins we currently have and maintain.

Reply:
I would be on the same side as @TurboTurtle, and see how we can keep common stuff in the same plugin.

As an example, we created the `microk8s` plugin, and amended that only to collect `microk8s` specific stuff. But, we also trigger the `kubernetes` plugin, and update some the commands that are specific for `microk8s`. This ensures that we are not duplicating code, and if we change something in the common place, it can be helpful for all scenarios see #3501 as an example

Another example ins the `sunbeam` plugin PR #3586, where we are re-utlised the code in `ovn_host` and `openvswitch`, but changed a few variables depending on the fact we have the `openstack_hypervisor` (sunbeam) snap installed

Reply:
OK, I can go that route and implement this in a similar manner.

Reply:
Refactored the implementation and re-submitted the changes. We are now doing this similar to Microk8s + Kubernetes implementation. Also used the opportunity to implement those additional clean-ups as well.